### PR TITLE
Feature/implement subject

### PR DESCRIPTION
**Pull request by fmisir**

### CODE REVIEW

File: backend/src/main/java/org/eclipse/tractusx/semantics/registry/service/ShellService.java [hunk 1]
```diff
@@ -65,121 +74,139 @@ public Shell save(Shell shell) {
...
+    public List<String> findExternalShellIdsByIdentifiersByAnyMatch(Set<ShellIdentifier> shellIdentifiers) {
         List<String[]> keyValueCombinations = shellIdentifiers.stream().map(shellIdentifier -> new String[]{shellIdentifier.getKey(), shellIdentifier.getValue()}).collect(Collectors.toList());
-        return shellRepository.findExternalShellIdsByIdentifiersByAnyMatch(keyValueCombinations);
+        return shellRepository.findExternalShellIdsByIdentifiersByAnyMatch(keyValueCombinations, tenantAware.getTenantId(), owningTenantId);
     }
 
     @Transactional(readOnly = true)
-    public List<Shell> findShellsByExternalShellIds(Set<String> externalShellIds){
+    public List<Shell> findShellsByExternalShellIds(Set<String> externalShellIds) {
         return shellRepository.findShellsByIdExternalIsIn(externalShellIds);
```
Comment (LuLeRoemer) [inline backend/src/main/java/org/eclipse/tractusx/semantics/registry/service/ShellService.java:126]: No filtering required here?

File: backend/src/main/resources/static/aas-registry-openapi.yaml [hunk 2]
```diff
@@ -788,12 +788,12 @@ components:
             minLength: 1
             maxLength: 200
           # not supported for Catena-X
-          # subjectId:
-            # $ref: '#/components/schemas/Reference'
           value:
             type: string
             minLength: 1
             maxLength: 200
+          externalSubjectId:
```
Comment (LuLeRoemer) [inline backend/src/main/resources/static/aas-registry-openapi.yaml:795]: Does it make sense to further restrict what values can be inserted here? Do we know more about the structure of a BPN (e.g. lenght)? I know that we want to keep the tenant information independently of any specifics, but maybe it makes sense to make it configurable.

**Review by LuLeRoemer — COMMENTED** [review]
I was able to validate that the id filtering is working. However, I am wondering why it is not applied to all endpoints that return shells.

File: backend/src/main/java/org/eclipse/tractusx/semantics/registry/service/ShellService.java (hunk 1 above)
Comment (fmisir) [reply backend/src/main/java/org/eclipse/tractusx/semantics/registry/service/ShellService.java:126]: Good catch, it's is required here as well.

**Review by LuLeRoemer — APPROVED** [review]
LGTM

**Pull request merged**